earth_enterprise/src/common/khMTTypes.h

```diff
@@ -148,9 +151,173 @@ class khMTSet
     khLockGuard guard(mutex);
     return std::vector<T>(set.begin(), set.end());
   }
+  void clear(void) {
+    khLockGuard guard(mutex);
+    set.clear();
+  }
+  size_t size(void) {
+    khLockGuard guard(mutex);
```

Comment:
Minor: you might be able to get away with not using the mutex for the size function. If you call size while an insertion is happening I don't know if we care whether it gives the size before or after the insertion.